conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/processing/MetadataProcessor.scala

```diff
@@ -0,0 +1,167 @@
+package tech.cryptonomic.conseil.indexer.tezos.processing
+
+import akka.stream.ActorMaterializer
+import tech.cryptonomic.conseil.common.tezos.Tables
+import tech.cryptonomic.conseil.common.tezos.TezosTypes.Micheline
+import tech.cryptonomic.conseil.indexer.config
+import tech.cryptonomic.conseil.indexer.tezos.michelson.contracts.TokenContracts.Tzip16
+import tech.cryptonomic.conseil.indexer.tezos.{
+  TezosGovernanceOperations,
+  TezosIndexedDataOperations,
+  TezosNamesOperations,
+  TezosNodeOperator,
+  Tzip16MetadataJsonDecoders,
+  Tzip16MetadataOperator,
+  TezosDatabaseOperations => TezosDb
+}
+import cats.instances.future._
+import cats.syntax.applicative._
+import cats.implicits._
+import tech.cryptonomic.conseil.common.tezos
+import slick.jdbc.PostgresProfile.api._
+import tech.cryptonomic.conseil.common.io.Logging.ConseilLogSupport
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+class MetadataProcessor(
```

Comment:
A comment explaining the purpose of this class and how it works would be very useful here.